#including <windows.h> emits warnings from driverspecs.h and no_sal2.h

At least with recent Windows SDKs (I tried 10.0.26100.0 and 10.0.22621.0), including windows.h from a C or header file emits
```
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(381): warning C4005: '_Kernel_acquires_resource_': macro redefinition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(876): note: see previous definition of '_Kernel_acquires_resource_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(391): warning C4005: '_Kernel_releases_resource_': macro redefinition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(880): note: see previous definition of '_Kernel_releases_resource_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(418): warning C4005: '_Kernel_requires_resource_held_': macro redefinition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(884): note: see previous definition of '_Kernel_requires_resource_held_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(439): warning C4005: '_Kernel_requires_resource_not_held_': macro redefinition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(888): note: see previous definition of '_Kernel_requires_resource_not_held_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(501): warning C5106: macro redefined with different parameter names
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(812): note: see previous definition of '_IRQL_raises_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(507): warning C5106: macro redefined with different parameter names
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(816): note: see previous definition of '_IRQL_requires_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(513): warning C5106: macro redefined with different parameter names
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(820): note: see previous definition of '_IRQL_requires_max_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(519): warning C5106: macro redefined with different parameter names
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(824): note: see previous definition of '_IRQL_requires_min_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(532): warning C5106: macro redefined with different parameter names
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(832): note: see previous definition of '_IRQL_saves_global_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(545): warning C5106: macro redefined with different parameter names
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(840): note: see previous definition of '_IRQL_restores_global_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(552): warning C5106: macro redefined with different parameter names
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(844): note: see previous definition of '_IRQL_always_function_min_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(558): warning C5106: macro redefined with different parameter names
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(848): note: see previous definition of '_IRQL_always_function_max_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(588): warning C4005: '_Kernel_clear_do_init_': macro redefinition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(892): note: see previous definition of '_Kernel_clear_do_init_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(594): warning C4005: '_Kernel_IoGetDmaAdapter_': macro redefinition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(896): note: see previous definition of '_Kernel_IoGetDmaAdapter_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(724): warning C4005: '_Kernel_float_saved_': macro redefinition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(864): note: see previous definition of '_Kernel_float_saved_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(730): warning C4005: '_Kernel_float_restored_': macro redefinition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(868): note: see previous definition of '_Kernel_float_restored_'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\driverspecs.h(737): warning C4005: '_Kernel_float_used_': macro redefinition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\no_sal2.h(872): note: see previous definition of '_Kernel_float_used_'
```

--- a/compiler/test/compilable/test22048.c
+++ b/compiler/test/compilable/test22048.c
@@ -1,0 +1,9 @@
+/*
+TEST_OUTPUT:
+---
+---
+*/
+
+#ifdef _MSCVER
+#include <windows.h>
+#endif

--- a/druntime/src/importc.h
+++ b/druntime/src/importc.h
@@ -158,10 +158,9 @@ typedef unsigned long long __uint64_t;
 #define _stdcall __stdcall
 #define _declspec __declspec
 
-// This header disables the Windows API Annotations macros
-// Need to include sal.h to get the pragma once to prevent macro redefinition.
-#include "sal.h"
-#include "no_sal2.h"
+// disable the Microsoft Source Code Annotation macros in "sal.h"
+#define _USE_DECLSPECS_FOR_SAL 0
+#define _USE_ATTRIBUTES_FOR_SAL 0
 #endif
 
 /****************************

